packages/core/typings/src/types/core/plugins/config/strapi-server/controllers.ts

```diff
@@ -0,0 +1,12 @@
+import type { Context } from 'koa';
+import { Strapi } from '../../../../..';
+
+export type ControllerReturnType = {
+  [key: string]: (ctx: Context) => Promise<unknown> | unknown;
+};
+
+export type Controller = ({ strapi }: { strapi: Strapi }) => ControllerReturnType;
```

Comment:
We shouldn't be using our own type here, we should be using the same type used by createCoreController https://github.com/strapi/strapi/blob/361b318f91416ef5984112ac5627eaff7988d750/packages/core/strapi/src/factories.ts#L22-L27

Reply:
This way it adds the strapi callback and also ensures that it only includes the extendable methods

Reply:
am not sure I understand we don't have uid here

Reply:
we don't have any predefined controllers or extendables,
plugin creators can define what they want

Reply:
Ah, true, sorry. But we're still re-inventing the wheel a bit. I'll take a look in the morning, it feels like there's something we can do.